### DISCUSSION TITLE
Can I check the permissions of a token

### DISCUSSION THREAD

**Discussion by tombrus** [community/community#25259] [Actions]
<p>Is it possible to get a list of the permissions of a given token?<br>
Or check if a token has a certain set of permissions?</p>
<p>In one of my actions I would like to warn that the provided token is lacking a specific permission. The errors currently issued by just trying to do something and failing are not always easy to understand. If I could check the permission before failing I could issue a helpful warning like <code>the provided token is missing the xxx permission</code>.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/tombrus">@tombrus</a>,</p>
<p>We seem have no available methods to list all the access scopes of a token.<br>
I tried using the <a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/oauth-authorizations" rel="noopener nofollow ugc"><strong>OAuth Authorizations API</strong></a>, but did not work.</p>
<p>If you are the authenticated user of the token, you can try to navigate to <strong>Personal settings</strong> &gt; <strong>Developer settings</strong> &gt; <strong>Personal access tokens</strong> to view the access scopes of the tokens you have created.</p>

**Comment by tombrus** [comment]
<p>Hi <a class="mention" href="/brightran">@brightran</a>!</p>
<p>interactively going to the token def will work, but it does not help in a script (unless you do some serious curl hacking I guess ;-)).</p>
<p>So this seems like a request for a valuable extension to the API, right?<br>
Can I file it somewhere?</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/tombrus">@tombrus</a>,</p>
<p>You can share a feature request in the <a href="https://support.github.com/contact/feedback" rel="noopener nofollow ugc"><strong>Feedback form for GitHub</strong></a>.<br>
That will allow you to directly interact with the appropriate engineering team, and make it more convenient for the engineering team to collect and categorize your suggestions.</p>

**Comment by tombrus** [comment]
<p>Thanks <a class="mention" href="/brightran">@brightran</a>, I entered a feature request.</p>

**Comment by henrik242** [comment, marked as answer]
<p>It’s fairly easy to check the permissions of a token using the github API:</p>
<pre><code class="lang-auto">$ curl -sS -f -I -H "Authorization: token ghp_yourOwnToken" https://api.github.com | grep -i x-oauth-scopes

x-oauth-scopes: public_repo, repo:status, repo_deployment
</code></pre>

**Reply by ben-elttam** [reply]
Works for classic PATs but not for fine-grained (beta) PATs.

**Comment by tombrus** [comment]
<p>Great henrik242! Thanks.</p>

**Comment by flxw** [comment]
Hi @tombrus and everybody else  - it's been quite some time since this was raised.
What's the best way to do this in 2023?
I am trying to accomplish this for the new fine-grained PATs that are provided via the `${github.token}` context in actions.

Is the feature request still open and can I support it somewhere?

**Reply by gautamkrishnar** [reply]
This is true, i had created a new bug https://github.com/orgs/community/discussions/73397

**Comment by FollowTheProcess** [comment]
FYI if you have the GitHub CLI installed, this is also possible with the `gh auth status` command, which you should be able to force to use a particular token via the `$GITHUB_TOKEN` env var, so something like:

```shell
GITHUB_TOKEN=<your_token> gh auth status
```

Seems like it would work to me

**Reply by vit-zikmund** [reply]
Today, this doesn't work as desired with the fine-grained tokens:
```  
$ GITHUB_TOKEN=$GH__FINE_READ gh auth status
github.com
  ✓ Logged in to github.com as vit-zikmund (GITHUB_TOKEN)
  ✓ Git operations for github.com configured to use https protocol.
  ✓ Token: github_pat_***********************************************************
```
This is the output with the classic one:
```
GITHUB_TOKEN=$GH_CLASSIC_REPO gh auth status
github.com
  ✓ Logged in to github.com as vit-zikmund (GITHUB_TOKEN)
  ✓ Git operations for github.com configured to use https protocol.
  ✓ Token: ghp_************************************
  ✓ Token scopes: read:org, repo
```

**Comment by axi92** [comment]
How does that work for installation app tokens?
